### PR TITLE
Add yq support

### PR DESCRIPTION
**Pull request by axhav**
Hi!
This is pull requests adds support for the yq (https://github.com/mikefarah/yq) yaml command line processor. 
Its a tool written in go, with similar functionallity to jq, so its nice to have on dependency restricted platforms.

### CODE REVIEW

File: doc/ale-yaml.txt [hunk 1]
```diff
@@ -366,5 +366,37 @@ g:ale_yaml_gitlablint_options                     *g:ale_yaml_gitlablint_options
...
+
+  wget https://github.com/mikefarah/yq/releases/download/${VERSION}/${BINARY}.tar.gz -O - | tar xz && mv ${BINARY} /usr/bin/yq
+
+Options
+-------------------------------------------------------------------------------
+
+g:ale_yaml_yq_executable                               *g:ale_yaml_yq_executable*
+                                                       *b:ale_yaml_yq_executable*
+  Type: |String|
+  Default: `'gll'`
```
Comment (hsanson) [inline doc/ale-yaml.txt:388]: This does not look right? should this be `yq` instead of `gll`?

File: doc/ale-yaml.txt [hunk 2]
```diff
@@ -366,5 +366,37 @@ g:ale_yaml_gitlablint_options                     *g:ale_yaml_gitlablint_options
...
+
+  This variable can be set to change the path to gll.
+
+
+g:ale_yaml_yq_options                                     *g:ale_yaml_yq_options*
+                                                          *b:ale_yaml_yq_options*
+  Type: |String|
+  Default: `''`
+
+  This variable can be set to pass additional options to gll.
```
Comment (hsanson) [inline doc/ale-yaml.txt:398]: Same here, should this be `yq` instead of `gll`?

File: doc/ale-yaml.txt [hunk 3]
```diff
@@ -366,5 +366,37 @@ g:ale_yaml_gitlablint_options                     *g:ale_yaml_gitlablint_options
...
+
+
```
Comment (hsanson) [inline doc/ale-yaml.txt:400]: What about `g:ale_yaml_yq_filters`? would be nice to document it.

**Review by hsanson — CHANGES_REQUESTED** [review]
Please add a tests for the Handle method. Good example would be `test/handler/test_jq_handler.vader `

**Comment by axhav** [comment]
Hey, sorry for taking some time. I've updated the PR, removed the copy-paste errors and added a test case, care to take a new look.

File: doc/ale-yaml.txt (hunk 1 above)
Comment (axhav) [reply doc/ale-yaml.txt]: Copy-paste issue, should be fixed now

**Review by hsanson — APPROVED** [review]
Thanks, LGTM.

**Pull request merged**